cpp/nacjd.cc

```diff
@@ -0,0 +1,360 @@
+    if (FileUtil::Exists(nacjd_titles))
+        FileUtil::DeleteFile(nacjd_titles);
+    
+    if (not Download(DOWNLOAD_URL, nacjd_titles, timeout * 1000))
+        LOG_ERROR("Could not download website with nacjd ids.");
+    std::ifstream file(nacjd_titles);
+    if (file)
+        std::for_each(std::istream_iterator<char>(file),
+                      std::istream_iterator<char>(),
+                      HandleChar);
+    else
+        LOG_ERROR("couldn't open file: " + nacjd_titles);
+    
+    if (FileUtil::Exists(nacjd_new_titles_json))
+        FileUtil::DeleteFile(nacjd_new_titles_json);
+    std::ofstream json_new_titles(nacjd_new_titles_json);
+    json_new_titles << "{ \"nacjd\" : [ " << "\n";
+    bool first(true);
+    for (const auto &id : ids_website) {
+        if (parsed_marc_ids.contains(id))
+            continue;
+        bool success = DownloadID(json_new_titles, id, /*use_separator*/ not first);
+        if (first and success)
+            first = false;
+        if (success)
+            ++(*number_of_new_ids);
+    }
+    json_new_titles << " ] }";
+}
+
+
+void ParseJsonAndWriteMarc(MARC::Writer * const title_writer) {
+    std::string json_document;
+    FileUtil::ReadStringOrDie(nacjd_new_titles_json, &json_document);
+    JSON::Parser json_parser(json_document);
+    std::shared_ptr<JSON::JSONNode> internal_tree_root;
+    if (not json_parser.parse(&internal_tree_root))
+        LOG_ERROR("Could not properly parse \"" + nacjd_new_titles_json + ": " + json_parser.getErrorMessage());
+
+    const auto root_node(JSON::JSONNode::CastToObjectNodeOrDie("tree_root", internal_tree_root));
+    int no_total(0), no_title(0), no_description(0), no_license(0), no_initial_date(0), no_keywords(0), no_creators(0);
+
+    std::shared_ptr<JSON::ArrayNode> nacjd_nodes(JSON::JSONNode::CastToArrayNodeOrDie("nacjd", root_node->getNode("nacjd")));
+    for (const auto &internal_nacjd_node : *nacjd_nodes) {
+        ++no_total;
+        bool complete = true;
+        std::string description;
+        std::string license;
+        std::string initial_release_date;
+        std::set<std::string> keywords;
+        std::map<std::string, std::string> creators;
+        const auto nacjd_node(JSON::JSONNode::CastToObjectNodeOrDie("entry", internal_nacjd_node));
+        const auto alternateIdentifiers_node(nacjd_node->getArrayNode("alternateIdentifiers"));
+        const auto distributions_node(nacjd_node->getArrayNode("distributions"));
+        const auto keywords_node(nacjd_node->getOptionalArrayNode("keywords"));
+        const auto creators_node(nacjd_node->getArrayNode("creators"));
+        const auto description_node(nacjd_node->getOptionalStringNode("description"));
+        const auto title_node(nacjd_node->getStringNode("title"));
+        if (description_node == nullptr)
+            complete = false;
+        else {
+            description = description_node->getValue();
```

Comment:
Declare "description" here!

Reply:
description is also used outside this scope, so cannot be delcared here.